python-sdk/src/astro/files/base.py

```diff
@@ -95,7 +95,13 @@ def create_from_dataframe(self, df: pd.DataFrame, store_as_dataframe: bool = Tru
             delimited data (e.g. csv, parquet, etc.).
         :param df: pandas dataframe
         """
+        from urllib.parse import urlparse
+
         self.is_dataframe = store_as_dataframe
+        parsed_url = urlparse(self.path)
+        if parsed_url.scheme == constants.FileLocation.FTP.value:
+            # For FTP we need to have append uri to path since it contains details of host, port, username, and password
+            self.path = f"{self.location.get_uri()}/{parsed_url.netloc}{parsed_url.path}"  # type: ignore [attr-defined]
```

Comment:
Can we not implement this somehow in ftp specific file path? Currently, `path` is a string, maybe we can have path specific to location. WDYT?

Reply:
Yes here also `path` is string itself but when user give he would have give only path we add uri which is a string which is used for smart_open for authentication.

Reply:
By default hostname , username and password isn't provided in `path`. So using uri i am appending into the path before using smart_open to write into the path

Reply:
@sunank200 I think we can fix this issue in the UTO implementation as we have location classes there where we could implement the specific path.

Reply:
currently we have moved to location.get_stream for each location and anything specific changes needed can be made in locations itself.